EIPS/eip-5170.md

```diff
@@ -0,0 +1,144 @@
+
+### Specification
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY” and “OPTIONAL” in this document are to be interpreted as described in [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+#### Format of the certificate and signature
+
+The certificate for the *script signing key* MUST be in the X.509 format, in accordance with [RFC 5280](https://datatracker.ietf.org/doc/html/rfc5280).
+Furthermore, the certificate MUST by signed by the *smart contract author* as the issuer, using the *smart contract deployment key*.
+
+We furthermore make the following requirements of the content of this certificate:
+
+- The `issuer` field MUST be populated with a Common Name, which MUST be the address of the verification key associated with the *smart contract deployment key*. E.g. `CN=0x12345678901234567890`.
+
+- The `SubjectPublicKeyInfo` field MUST contain the public part of the *script signing key*.
+
+- The `extensions` field SHOULD be set to include `KeyUsage` (see RFC 5280 sec. 4.2.1.3) with bit 0 set, to indicate that the *script signing key* is used for signing only. Furthermore the *Extended Key Usage* extensions SHOULD also be included, with only the *id-kp-codeSigning* identifier set (See RFC 52080 sec. 4.2.1.12).
+
+- If [revocation option 1](#Revocation) is used, then `extensions` MUST also include `cRLDistributionPoints` (see RFC 5280 4.2.1.13) which MUST contain at least one `distributionPoint` element, containing a `fullName` element. The `fullName` MUST be a single `IA5String` for a `uniformResourceIdentifier` which is an URI pointing to a Certificate Revocation List.
+
+- The `notBefore` and `notAfter` field SHOULD be set to limit lifetime of the *script signing key* reasonably.
+
+- The `version` field SHOULD be grater than 2, to indicate that the certificate is **not** a regular X.509 certificate.
+
+
+We require the signature to be done as a JWS according to [RFC 7515](https://datatracker.ietf.org/doc/html/rfc7515). Concretely we have the following requirements to the elements contained in the JWS:
+
+- The `x5u` header MUST be included and stored as part of the `JWS Protected Header`. Furthermore, its URI MUST point to the X.509 certificate of the *script signing key*.
+
+- The `payload` member MUST be exactly the URL base64 encoding of the client script *or* an a Keccak of the client script.
+
+
+#### Format to attach signature
+
+This ERC does not specify how a wallet client obtains the signature as this can be realized in multiple ways. The simplest of which is to embed the client script in the `payload` of the JWS, as discussed above. This ensure that only a single URI is needed in order to locate both the client script, its signature and the signing key certificate. 
```

Comment:
```suggestion
This EIP does not specify how a wallet client obtains the signature as this can be realized in multiple ways. The simplest of which is to embed the client script in the `payload` of the JWS, as discussed above. This ensure that only a single URI is needed in order to locate both the client script, its signature and the signing key certificate. 
```
```suggestion
This ERC does not specify how a wallet client obtains the signature as this can be realized in multiple ways. The simplest of which is to embed the client script in the `payload` of the JWS, as discussed above. This ensure that only a single URI is needed in order to locate both the client script, its signature and the signing key certificate. 
```